numpy/core/fromnumeric.py

```diff
@@ -52,6 +52,39 @@ def _wrapit(obj, method, *args, **kwds):
     return result
 
 
+def _is_compat_error(e, method):
+    """
+    Checks whether the error raised from calling a method
+    or function in numpy is actually a compatibility error.
+
+    Parameters
+    ----------
+    e : Exception
+        The exception or error raised when calling the
+        method or function.
+    method : str
+        The name of the function or method that was called
+
+    Returns
+    -------
+    is_compat_error : bool
+        Indicates whether or not the error raised was in fact
+        a compatibility error with numpy
+    """
+    msg = str(e)
+
+    attr_err_msg = "has no attribute '{func}'"
+    type_err_msg = "{func}() got an unexpected keyword argument"
+
+    if (type(e) == AttributeError and
```

Comment:
My `flake8` disagrees unfortunately.  This is a moot point though, as I am going to decouple as you suggested.